addons/project/models/res_partner.py

```diff
@@ -80,3 +80,30 @@ def action_view_tasks(self):
         else:
             action['domain'] = search_domain
         return action
+
+    @api.model
+    def get_mention_suggestions_from_task(self, task_id, search, limit=8):
+        """Return 'limit'-first partners' such that the name or email matches a 'search' string.
+        Prioritize partners that are also (internal) users, and then extend the research to all partners.
+        Only followers of the given task or followers of its project are returned.
+        The return format is a list of partner data (as per returned by `mail_partner_format()`).
+        """
+        task_sudo = self.env['project.task'].sudo().search([('id', '=', task_id)])
+        shared_project_sudo = task_sudo.project_id
```

Comment:
We should make sure the user has access to the task, to do that, we can remove the sudo on task. 🙂 
Also, since we have the id, no need to do a search, you can call browse method instead. 🙂 

```suggestion
        task = self.env['project.task'].browse(task_id)
        project = task.project_id
```

However, if we apply my comment above, then we no longer need to browse the recordset. But we can make sure we only have one record contains in `self`. 🙂 

```suggestion
        self.ensure_one()
        project = self.project_id
```